.github/workflows/macos.yml

```diff
@@ -72,11 +73,6 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0 # fetch everything for python setuptools_scm
-      - name: Download libtiledbvcf artifact
```

Comment:
This step shouldn't be removed. The whole point of this `python` job that depends on the previous `libtiledbvcf` job is to test that python can be built against an existing libtiledbvcf. The job `python-standalone` below already tests the ability of Python to bootstrap a libtiledbvcf installation. We want to test that this continues to work (since it is required eg for our conda feedstock builds).